Sources/SwiftSyntax/generated/SyntaxCollections.swift

```diff
@@ -1424,14 +1504,59 @@ public struct StringLiteralSegmentListSyntax: SyntaxCollection, SyntaxHashable {
       }
     }
 
+    /// Creates a new Element wrapping a string segment.
+    ///
+    /// This initializer is used when adding literal text content to a string literal. The string segment should be created using `.stringSegment()` to ensure proper formatting.
+    ///
+    /// Example from warning macro:
+    /// ```swift
+    /// let element = StringLiteralSegmentListSyntax.Element(
+    ///   StringSegmentSyntax(content: .stringSegment("warning: invalid configuration"))
+    /// )
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - node: The string segment to wrap in this element.
     public init(_ node: StringSegmentSyntax) {
       self = .stringSegment(node)
     }
 
+    /// Creates a new Element wrapping an expression segment.
+    ///
+    /// This initializer is used when adding interpolated expressions to a string literal.
+    ///
+    /// Example from diagnostic message:
+    /// ```swift
+    /// let element = StringLiteralSegmentListSyntax.Element(
+    ///   ExpressionSegmentSyntax(
+    ///     expressions: ExprListSyntax([
+    ///       ExprSyntax("errorValue")
+    ///     ])
+    ///   )
+    /// )
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - node: The expression segment to wrap in this element.
     public init(_ node: ExpressionSegmentSyntax) {
       self = .expressionSegment(node)
     }
 
+    /// Creates a new Element from an existing syntax node by attempting to cast it to either a string or expression segment.
+    ///
+    /// Example from string literal parsing:
```

Comment:
Similar to my comments above, I would prefer to keep this a standalone comment without referencing where it’s from.